src/Symfony/Component/Serializer/Encoder/XmlEncoder.php

```diff
@@ -37,16 +37,19 @@ class XmlEncoder implements EncoderInterface, DecoderInterface, NormalizationAwa
     private $context;
     private $rootNodeName = 'response';
     private $loadOptions;
+    private $ignoredNodeTypes;
 
     /**
      * Construct new XmlEncoder and allow to change the root node element name.
      *
-     * @param int|null $loadOptions A bit field of LIBXML_* constants
+     * @param int|null $loadOptions      A bit field of LIBXML_* constants
+     * @param int[]    $ignoredNodeTypes an array of ignored XML node types, each one of the DOM Predefined XML_* Constants
      */
-    public function __construct(string $rootNodeName = 'response', int $loadOptions = null)
+    public function __construct(string $rootNodeName = 'response', int $loadOptions = null, array $ignoredNodeTypes = null)
```

Comment:
```php
, array $ignoredNodeTypes = array(XML_PI_NODE, XML_COMMENT_NODE)
```
then you can get rid of ternary

Reply:
I wondered about that. I was following the pattern I saw for `$loadOptions`, but I'm happy to switch to that.